### PR TITLE
Issue 45: Throw exception if profile doesn't exist

### LINKED ISSUE #45: Throw an exception when the profile passed in does not exist

### PR DESCRIPTION
**Pull request by steveworley**
- Updates Profile class so that it only pertains to a "Profile"
- Adds a ProfileController class to abstract loading/saving
- Updates Audit commands so that they now load via the controller
- Updates HtaccessRedirects to prevent errors when generating a profile

Fixes #45

### CODE REVIEW

**Comment by steveworley** [comment]
@seanhamlin I created #52 off of this - I had to update the `ProfileCreateCommand` to go with a more MVC approach for the Profile class and when I was testing I found issues with the `array_flip` that the Profile class does when it is adding a check.

File: src/Profile/Profile.php [hunk 1]
```diff
@@ -121,6 +118,7 @@ public function getMachineName()
   public function addCheck($check, $options = array())
   {
     $registry = Registry::load();
+    file_put_contents(json_encode($registry), '/Users/steven.worley/repos/site-audit/file.txt');
```
Comment (seanhamlin) [inline src/Profile/Profile.php]: looks like it is debug.

File: src/Profile/ProfileController.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  public static function find($machine_name) {
+    $filename = $machine_name . '.yml';
+    foreach (array_filter(self::FILE_PATHS, 'is_dir') as $file_path) {
+      $location = $file_path . '/' . $filename;
+      if (file_exists($location)) {
+        return $location;
+      }
+    }
+
+    throw new ProfileNotFoundException($machine_name, 0);
```
Comment (seanhamlin) [inline src/Profile/ProfileController.php]: w00t

**Review by seanhamlin — CHANGES_REQUESTED** [review]

**Comment by seanhamlin** [comment]
Nice work @steveworley !

**Pull request merged**